SupportBank/Program.cs

```diff
@@ -1,46 +1,47 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Globalization;
-using System.IO;
-using CsvHelper;
+using System.Linq;
 
 namespace SupportBank
 {
     class Program
     {
-        private static Dictionary<string,Account> _accounts;
-        
         static void Main(string[] args)
         {
-            _accounts = new Dictionary<string, Account>();
-            CSVReader();
-            foreach (var account in _accounts.Values)
+            var bs = new Base();
+            bs.ParseFileAndPopulateDictionary();
+            while (true)
             {
-                Console.WriteLine($"Name: {account.Name} -- Pending Account Balance: {account.BalanceDue}");
-            }
-        }
+                Console.WriteLine("Welcome to the Support Bank. Type 'List All' to see all balances or 'List [Account]' to see the transactions of a particular account. Type 'exit' to exit ");
+                try
+                {
+                    var commands = Console.ReadLine().Split(" ");
+                    if (commands[0].ToLower() == "exit")
+                    {
+                        break;
+                    }
 
-        static void CSVReader()
-        {
-            using (var reader = new StreamReader("./Transactions2014.csv"))
-            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
-            {
-                var records = csv.GetRecords<Transaction>();
-                foreach (var transaction in records)
+                    if (commands[0].ToLower() == "list")
+                    {
+                        if (commands[1].ToLower() == "all")
+                        {
+                            bs.ListAll();
+                        }
+                        else
+                        {
+                            bs.ListAccount(String.Join(" ", commands.Skip(1)));
+                        }
+                    }
+
+                    else
+                    {
+                        throw new Exception();
+                    }
+                }
+                catch (Exception)
```

Comment:
Which exceptions are you expecting to occur, other than the one you throw yourself? 🙂

Reply:
There's a case in which someone could just type the word "list". In that case commands would have only one element and commands[1] would get an index out of bounds exception

Reply:
I see. In that case I would change
```
if (commands[0].ToLower() == "list")
```
to
```
if (commands[0].ToLower() == "list" && commands.length > 1)
```
or similar so that you can catch it before the exception occurs. That would make the code more readable, and means you don't need a try/catch 🙂